### DISCUSSION TITLE
Version 0.50.0 complex templates may be broken

### DISCUSSION THREAD

**Discussion by oPromessa** [RhetTbull/osxphotos#708] [Q&A] [answered]
Hey there. The read me for Version 0.50.0 mentions:

There may be breaking changes if you use complex templates.

The primary change is that the PATH_SEP argument must now come before any filters.

==> question: how can one tell if is using a complex template? Would these be complex? Maybe not!
```
keyword_template = [ "{keyword}", "{label}", "{searchinfo.activity?activity:{searchinfo.activity},}", "{searchinfo.venue_type?venue_type:{searchinfo.venue_type},}", ]
```

```
description_template = "Album:{album,}{newline}Description:{descr,}"
```

**Comment by RhetTbull** [comment]
Those should be fine. You'll know if it's broken because you'll get a syntax error. I think the only thing that will really break is folder_album + path_sep in conjunction with filter where the order needs to be reversed.

**Comment by RhetTbull** [comment, marked as answer]
@oPromessa In v0.50.3 I've added a `--template/-T` option to the `osxphotos inspect` command.  This allows you to preview templates for photos selected in Photos and should be useful for testing new templates.  For example, for your templates, you can do this:

`osxphotos inspect -T "{keyword}" -T "{label}" -T "{searchinfo.activity?activity:{searchinfo.activity},}" -T "{searchinfo.venue_type?venue_type:{searchinfo.venue_type},}"`

Then select a photo in Photos and the terminal window will automatically render those templates for the selected photo.
